### PR TITLE
Bigger refactor to better clarify responsibility.

### PR DESCRIPTION
**Pull request by rejdi**
Greatly improved performance for many selected countries.

Sorry guys for this bigger one, but it was necessary and I couldn't break it down into smaller pieces.

### CODE REVIEW

File: web/react-web/src/Content/Graph/Graph.js [hunk 1]
```diff
@@ -69,6 +69,16 @@ function xaxis_text(x, isXAxisRelative) {
     }
 }
 
+function generateNumber(value) {
```
Comment (lukipuki) [inline web/react-web/src/Content/Graph/Graph.js]: Is this a random number generator? Or a hash calculation?

It seems it's the latter, so please call the function `calculateHash` or similar.

File: web/react-web/src/Commons/functions.js [hunk 2]
```diff
@@ -1 +1 @@
-export const isSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value));
\ No newline at end of file
+export const isSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value));
```
Comment (lukipuki) [inline web/react-web/src/Commons/functions.js]:
```suggestion
export const areSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value));
```

**Review by lukipuki — APPROVED** [review]
Can't review too much of the JS but if it works, ship it!

File: web/react-web/src/Content/Graph/Graph.js (hunk 1 above)
Comment (rejdi) [reply web/react-web/src/Content/Graph/Graph.js]: Applied.

File: web/react-web/src/Commons/functions.js (hunk 2 above)
Comment (rejdi) [reply web/react-web/src/Commons/functions.js]: Applied.

**Pull request merged**